test/test_optim.py

```diff
@@ -315,15 +316,20 @@ def closure():
 
             self.assertLess(closure().item(), initial_value)
 
+    @parametrize("num_dim", [0, 1, 2, 3])
```

Comment:
is there a diff between 2 and 3, do you think?
```suggestion
    @parametrize("num_dim", [0, 1, 3])
```
would this be the same in terms of coverage?

Reply:
2 is large enough to cause one like the following error because `params` contains a 1-dim tensor (`bias`) in `test_tensor_lr`.

```
output with shape [2] doesn't match the broadcast shape [1, 2]
```